### ISSUE TITLE
Show help text

### ISSUE THREAD

**Issue by MartinHelmut** [krieselreihe/litr#25]
Currently the special parameter `-h` / `--help` does nothing. Goal is that this will print a beautiful auto generated help text inside the terminal.

Reference material:
- http://docopt.org/
- https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/command-line-syntax-key
- https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html#tag_12_01
- https://clig.dev/ (🥰)

**Comment by MartinHelmut** [comment]
First idea for output:

```shell
Litr - Language Independent Task Runner [version 1.0.0]
  Configuration file found under: /Users/martin/Projects/krieselreihe/litr/litr.toml

Usage:
  test   [--target=<argument>]
  build  [--target=<argument>]
  run    [--target=<argument>]
  update

Options:
  -h --help    Show this screen.
  -t --trace   Activate tracing support.
  -t --target  Define the application build target.
    One of: "debug", "release", "profile", "trace"
```

**Comment by MartinHelmut** [comment]
First workable draft:

```shell
Litr - Language Independent Task Runner [version 1.0.0]
  Configuration file found under: /Users/martin/Projects/krieselreihe/litr/litr.toml

Usage: litr command [options]

Commands:
  test    [--target=<option>]
  build   [--trace] [--target=<option>]
  run     [--target=<option>]
  update

Options:
  -h --help             Show this screen.
     --something        Lorem ipsum dolor sit amet, consectetur adipiscing elit,
                        sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut
                        enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi.
  -t --trace            Activate tracing support.
  -a --target=<option>  Define the application build target.
                        Available options: "debug", "release", "profile", "trace"
                        Default option is: "debug"
```

**Comment by MartinHelmut** [comment]
Currently there is only the global help available (`litr --help`) but not a command specific. Given the following execution:

```shell
litr run --help
```

This should be the expected help:

```shell
Litr - Language Independent Task Runner [version 1.0.0]
  Configuration file found under: /Users/martin/Projects/krieselreihe/litr/litr.toml

Usage: litr run [options]

Options:
  -h --help             Show this screen.
  -t --target=<option>  Define the application build target.
                        Available options: "debug", "release", "profile", "trace"
                        Default option is: "debug"
```

**Comment by MartinHelmut** [comment]
Command help is implemented with 09f39c5dd6a92357313f5992210329bc8de5a1a0 . Questions remains, how to show help for nested commands.

**Comment by MartinHelmut** [comment]
Nested command help idea:

```shell
Litr - Language Independent Task Runner [version 1.0.0]
  Configuration file found under: /Users/martin/Projects/krieselreihe/litr/litr.toml

Usage: litr command [options]

Commands:
  test    [--target=<option>]
    cpp   [--target=<option>]
    java  [--target=<option>]
  build   [--trace] [--target=<option>]
  run     [--target=<option>]
  update

Options:
  -h --help             Show this screen.
     --something        Lorem ipsum dolor sit amet, consectetur adipiscing elit,
                        sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut
                        enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi.
  -r --trace            Activate tracing support.
  -t --target=<option>  Define the application build target.
                        Available options: "debug", "release", "profile", "trace"
                        Default option is: "debug"
```
